threads.js

```diff
@@ -2036,14 +2036,32 @@ Process.prototype.reportProduct = function (a, b) {
     return +a * +b;
 };
 
+function QuotientNaNException(numerator, denominator) {
+    this.name = "Quotient exception";
+    this.message = "illegal division operation: (" + numerator + ") / (" + denominator + ")";
+}
+
 Process.prototype.reportQuotient = function (a, b) {
-    return +a / +b;
+    var result = +a / +b;
+    if (isNaN(result) || +b === 0) {
+        throw new QuotientNaNException(+a, +b);
+    }
+    return result;
 };
 
+function ModulusNaNException(numerator, denominator) {
+    this.name = "Modulus exception";
+    this.message = "illegal modulus operation: (" + numerator + ") / (" + denominator + ")";
```

Comment:
Perhaps we could should this as `num % denom` or `num mod denom` The `/` symbol seems confusing to me...